src/lib/wallets/useEthersSigner.ts

```diff
@@ -16,8 +17,33 @@ export function clientToSigner(client: Client<Transport, Chain, Account>) {
   return signer;
 }
 
+const cache = new Map<string, UncheckedJsonRpcSigner>();
+
 /** Hook to convert a Viem Client to an ethers.js Signer. */
 export function useEthersSigner({ chainId }: { chainId?: number } = {}) {
   const { data: client } = useConnectorClient<Config>({ chainId });
-  return useMemo(() => (client ? clientToSigner(client) : undefined), [client]);
+
+  const id = client?.uid;
+
+  useEffect(() => {
+    return () => {
+      if (id) {
+        cache.delete(id);
+      }
+    };
+  }, [id]);
+
+  if (!client || !id) {
+    return undefined;
+  }
+
+  if (cache.has(id)) {
+    return cache.get(id);
+  }
+
+  const signer = clientToSigner(client);
+
+  cache.set(id, signer);
```

Comment:
why we need a cache here? is clientToSigner that slow even useMemo not enough?

Reply:
reverted, not that big of a deal